internal/shim-sev/src/hostcall.rs

```diff
@@ -92,13 +91,13 @@ impl<'a> HostCall<'a> {
     /// The parameters returned can't be trusted.
     pub unsafe fn write(&mut self, fd: usize, bytes: &[u8]) -> sallyport::Result {
         let cursor = self.0.cursor();
-        let (_, buf) = cursor.copy_slice(bytes).or(Err(libc::EMSGSIZE))?;
+        let (_, buf) = cursor.copy_from_slice(bytes).or(Err(libc::EMSGSIZE))?;
 
         let buf_address = Address::from(buf.as_ptr());
         let phys_unencrypted = ShimPhysUnencryptedAddr::try_from(buf_address).unwrap();
         let host_virt: HostVirtAddr<_> = phys_unencrypted.into();
 
-        self.0.msg.req = request!(libc::SYS_write => fd, host_virt, buf.len());
+        self.0.msg.req = request!(libc::SYS_write => fd, host_virt, bytes.len());
```

Comment:
`buf.len()` and `bytes.len()` are equal, because using `Cursor::copy_from_slice()` uses `Cursor::alloc()`, which allocates `bytes.len()` with or fails.
```
        let (_, buf) = cursor.copy_from_slice(bytes).or(Err(libc::EMSGSIZE))?;
```

Anyway... I changed it back.